R/user.R

```diff
@@ -72,3 +72,47 @@ resetPassword <- function (email) {
             url_base=absoluteURL("../password/change/${token}/", app)
         ))))
 }
+
+#' @rdname catalog-extract
+#' @export
+setMethod("[", c("UserCatalog", "character"), function (x, i, ...) {
+    dots <- list(x=x, i=i, secondary = emails(x))
```

Comment:
Don't think you need `x` and `i` in this part, do you? They must be in `list(...)`.

Reply:
Unless I'm misunderstanding, `x` and `i` aren't in `list(...)` it's only all of the other args:

```
> foo <- function(x, i, ...) {
+     print(list(...))
+ }
> foo(x="x", i="i", secondary="sec")
$secondary
[1] "sec"
```

We also need them in the `setMethod` definition since we are dispatching on them, so we can't pull them from there so that they get passed with `...`.

Reply:
Right, of course, I must have had stuck in my brain the `match.call` stuff you were talking about last week. My bad.